src/Host/Client/Impl/Session/RSessionProvider.cs

```diff
@@ -81,7 +82,14 @@ public class RSessionProvider : IRSessionProvider {
                 return;
             }
 
-            foreach (var session in _sessions.Values) {
+            var sessions = GetSessions();
+            var stopHostTasks = sessions.Select(session => session.StopHostAsync());
+            try {
+                Task.WhenAll(stopHostTasks).GetAwaiter().GetResult();
```

Comment:
To gracefully stop the hosts when we close VS, instead of merely disconnecting from them (disposing an `RHost` instance disconnects from the broker, but doesn't tell it to terminate the process).